### PR TITLE
Use index data to render search template

### PR DESCRIPTION
**Pull request by sonalkr132**
This PR also adds  `latest_verison_number` to index because we need it for rendering.

It will render template for search *without* making following  queries(for 3 matches found):
```sql
Rubygem Load (1.3ms)  SELECT "rubygems".* FROM "rubygems" WHERE "rubygems"."id" IN (4, 97, 132)
   (0.6ms)  SELECT COUNT(count_column) FROM (SELECT  1 AS count_column FROM "rubygems" WHERE "rubygems"."name" = $1 LIMIT 1) subquery_for_count  [["name", "active"]]
  Rubygem Load (3.3ms)  SELECT  "rubygems".* FROM "rubygems" WHERE (UPPER(name) = UPPER('active')) AND (rubygems.id IN (SELECT rubygem_id FROM versions where versions.indexed IS true))  ORDER BY "rubygems"."id" ASC LIMIT 1
   (0.7ms)  SELECT COUNT(*) FROM "rubygems" WHERE "rubygems"."id" IN (4, 97, 132)
  CACHE (0.0ms)  SELECT COUNT(*) FROM "rubygems" WHERE "rubygems"."id" IN (4, 97, 132)
  CACHE (0.0ms)  SELECT COUNT(*) FROM "rubygems" WHERE "rubygems"."id" IN (4, 97, 132)
  CACHE (0.0ms)  SELECT "rubygems".* FROM "rubygems" WHERE "rubygems"."id" IN (4, 97, 132)
  Version Load (1.0ms)  SELECT  "versions".* FROM "versions" WHERE "versions"."rubygem_id" = $1 AND "versions"."latest" = $2 LIMIT 1  [["rubygem_id", 4], ["latest", "t"]]
  GemDownload Load (1.2ms)  SELECT  "gem_downloads".* FROM "gem_downloads" WHERE "gem_downloads"."rubygem_id" = $1 AND "gem_downloads"."version_id" = $2 LIMIT 1  [["rubygem_id", 4], ["version_id", 0]]
  Version Load (0.6ms)  SELECT  "versions".* FROM "versions" WHERE "versions"."rubygem_id" = $1 AND "versions"."latest" = $2 LIMIT 1  [["rubygem_id", 97], ["latest", "t"]]
  GemDownload Load (0.5ms)  SELECT  "gem_downloads".* FROM "gem_downloads" WHERE "gem_downloads"."rubygem_id" = $1 AND "gem_downloads"."version_id" = $2 LIMIT 1  [["rubygem_id", 97], ["version_id", 0]]
  Version Load (0.6ms)  SELECT  "versions".* FROM "versions" WHERE "versions"."rubygem_id" = $1 AND "versions"."latest" = $2 LIMIT 1  [["rubygem_id", 132], ["latest", "t"]]
  GemDownload Load (0.5ms)  SELECT  "gem_downloads".* FROM "gem_downloads" WHERE "gem_downloads"."rubygem_id" = $1 AND "gem_downloads"."version_id" = $2 LIMIT 1  [["rubygem_id", 132], ["version_id", 0]]
 ```

### CODE REVIEW

File: app/views/rubygems/_rubygem.html.erb [hunk 1]
```diff
@@ -1,10 +1,10 @@
-<%= link_to rubygem_path(rubygem), :class => 'gems__gem' do %>
+<%= link_to "/gems/#{rubygem.name}", :class => 'gems__gem' do %>
```
Comment (dwradcliffe) [inline app/views/rubygems/_rubygem.html.erb]: Why not use the route helper?

Comment (sonalkr132) [inline app/views/rubygems/_rubygem.html.erb]: route helper would only work on activerecord object and not the response returned by ES. We validate `name` against `[A-Za-z0-9.-_]`, so this shouldn't lead to any unintended behavior.

Comment (dwradcliffe) [inline app/views/rubygems/_rubygem.html.erb]: You can pass in the param instead of the activerecord object, so: `rubygem_path(rubygem.name)`.

Comment (sonalkr132) [inline app/views/rubygems/_rubygem.html.erb]: Sweet :cherries:

**Pull request merged**